### ISSUE TITLE
Document new staging workflow

### ISSUE THREAD

**Issue by worldofpeace** [NixOS/nixpkgs#83381]
**Describe the bug**
Since [RFC 26](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md) was merged we have a new staging workflow that involves a `staging-next` branch.
It was merged on 15 Nov 2018, but was implemented a bit before that.
Part of the `Future Work` was to document this workflow so new contributors can understand this. We still need to do this.

cc @vcunat @FRidh

**Comment by vcunat** [comment]
I expect the main thing people need are [those three bullets](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md#detailed-design), right?  Especially the first two ones, but the workflow change was specifically designed not to really change those two points, so that most people are as little affected as possible.

**Comment by vcunat** [comment]
Wait, 19.09 already has (some) docs for all this: https://nixos.org/nixpkgs/manual/#submitting-changes-staging-next-branch ... so can you (anyone) be more specific about what seems missing?

**Comment by worldofpeace** [comment]
> Wait, 19.09 already has (some) docs for all this: https://nixos.org/nixpkgs/manual/#submitting-changes-staging-next-branch ... so can you (anyone) be more specific about what seems missing?

Woah, wonder how I missed that. Added in https://github.com/NixOS/nixpkgs/commit/62193ec416a770925eb3bbe389a7480f76af92eb. Though I do think some things are missing because if both of you were to be unavailable the workflow to do the thing is unclear.
Like, from what I see there how are all the merges done?
I've figured out from observation that you
* merge staging into staging-next (begins staging-next process)
* pr to master
* avoid conflicts
  * merge master into staging-next
  * merge staging-next into staging
* merge staging-next into master (promote staging-next to master)

We could also document the details of the jobsets like it is in the RFC, and how is someone supposed to determine staging-next is of sufficient quality? I do understand this involves looking at hydra.

It's documented how to submit changes, but I think the workflow around it isn't.

I'm raising this because I'm getting the impression that maybe more people would like to help out getting staging-next PRs merged, so there is this concept in nixpkgs of the "regular staging-next PR".
